src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SqlManagerController.php

```diff
@@ -175,56 +172,33 @@ public function processFormAction(Request $request)
      */
     public function createAction(Request $request)
     {
-        $requestSqlFormHandler = $this->getRequestSqlFormHandler();
-
-        $requestSqlData = [];
-
-        // handle "Export to SQL Manager" action data
-        if ($request->request->has('sql')) {
-            $requestSqlData['request_sql'] = [
-                'sql' => $request->request->get('sql'),
-                'name' => $request->request->get('name'),
-            ];
-        }
-
-        $requestSqlForm = $requestSqlFormHandler->getForm();
-        $requestSqlForm->setData($requestSqlData);
-        $requestSqlForm->handleRequest($request);
+        $sqlRequestForm = $this->getSqlRequestFormHandler()->getForm();
+        $sqlRequestForm->handleRequest($request);
 
-        if ($requestSqlForm->isSubmitted()) {
-            if ($this->isDemoModeEnabled()) {
-                $this->addFlash('error', $this->getDemoModeErrorMessage());
-
-                return $this->redirectToRoute('admin_sql_request');
-            }
-
-            $errors = $requestSqlFormHandler->save($requestSqlForm->getData());
-
-            if (empty($errors)) {
-                $this->addFlash('success', $this->trans('Successful creation.', 'Admin.Notifications.Success'));
-
-                return $this->redirectToRoute('admin_sql_request');
-            }
+        if ($this->getSqlRequestFormHandler()->handle($sqlRequestForm)) {
```

Comment:
@matks @jolelievre do you think `handleRequest($request)` could be moved into `FormHandler::handle()`?

Reply:
Looks good in the controller to me 😄 it's a controller's job to manipulate the request object

Reply:
Re-edit: as suggested upper (sorry my comments are not sorted chronoligically), maybe we should let the controller extract the data and use `$form->submit($data)` in the `FormHandler`

Reply:
Honestly I think it's a detail 😄 both solutions I'm OK with that. However your solutions (will require to pass both `$request` (or `$data`) and `$form` to the FormHandler, instead of only `$form`, right ?

Reply:
yep, that's the drawback..